openmdao/core/fileref.py

```diff
@@ -0,0 +1,93 @@
+"""
+Support for file variables.
+"""
+
+import sys
+import copy
+import os
+
+#Public Symbols
+__all__ = ['FileRef']
+
+#_big_endian = sys.byteorder == 'big'
+
+CHUNK = 1 << 20  # 1MB
+
+class FileRef(object):
+    """
+    A reference to a file on disk. As well as containing metadata information,
+    it supports :meth:`open` to read and write the file's contents.
+    """
+
+    def __init__(self, fname, binary=False):
+        #, desc='', content_type='', platform=sys.platform,
+              #big_endian=_big_endian, single_precision=False,
+              #integer_8=False, unformatted=False, recordmark_8=False):
+        self.fname = fname
+        self.parent_dir = None
+        self.binary = binary
+        # self.desc = desc
+        # self.content_type = content_type
+        # self.platform = platform
+        # self.big_endian = big_endian
```

Comment:
Scott had all of those as metadata attributes in his file passing stuff.  I just left them in for now until I get done with the NPSS and AnalysisServer components, just in case I need them.